spec/unit/3scale_toolbox_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'fileutils'
+require 'shared_contexts'
+require '3scale_toolbox'
+
+RSpec.describe ThreeScaleToolbox do
+  include_context :temp_dir
+  include_context :plugin
+  include_context :random_name
+
+  let(:name) { random_lowercase_name }
+  let(:dest_plugin_file) { File.join(tmp_dir, '3scale_toolbox_plugin.rb') }
+
+  before(:each) do
+    plugin = get_plugin_content(name.capitalize, name)
+    File.open(dest_plugin_file, 'w') do |file|
+      file.write(plugin)
+    end
+    $LOAD_PATH.unshift(tmp_dir) unless $LOAD_PATH.include?(tmp_dir)
```

Comment:
I guess this should be in the `temp_dir` context. You are using it in both tests anyway.

Reply:
separation of concerns. Temporary directories might be useful not only for adding to load path.